server/src/main/java/org/opensearch/index/engine/InternalEngine.java

```diff
@@ -1005,10 +1005,18 @@ protected final IndexingStrategy planIndexingAsNonPrimary(Index index) throws IO
             assert maxSeqNoOfUpdatesOrDeletes < index.seqNo() : index.seqNo() + ">=" + maxSeqNoOfUpdatesOrDeletes;
             plan = IndexingStrategy.optimizedAppendOnly(index.version(), 0);
         } else {
+            boolean segRepEnabled = engineConfig.getIndexSettings().isSegRepEnabled();
             versionMap.enforceSafeAccess();
             final OpVsLuceneDocStatus opVsLucene = compareOpToLuceneDocBasedOnSeqNo(index);
             if (opVsLucene == OpVsLuceneDocStatus.OP_STALE_OR_EQUAL) {
-                plan = IndexingStrategy.processAsStaleOp(index.version());
+                if (segRepEnabled) {
+                    // For segrep based indices, we can't completely rely on localCheckpointTracker
+                    // as the preserved checkpoint may not have all the operations present in lucene
```

Comment:
> as the preserved checkpoint may not have all the operations present in lucene

Why would this be the case? Also I didn't understand how `processButSkipLucene` helps here(maybe needs more context).

Reply:
The checkpoint taken for Remote Store is  `indexShard.getEngine()).currentOngoingRefreshCheckpoint()` . Just after engine saves this checkpoint and between it actually starts refresh , operations can actually get written. Hence the ongoing checkpoint is just a marker which guarantees that refresh will have operations till that , but can have some more operations . 

`processButSkipLucene` : this doesn't index the document, but returns since  it has already been processed .